docs/building_docker_images.md

```diff
@@ -0,0 +1,154 @@
+# Building Docker Images
+
+## Cargo Features
+
+The Hyperswitch application server makes extensive use of
+[Cargo features][cargo-features] to selectively toggle certain features and
+integrations.
+The list of features can be found in the `[features]` section in the
+[`crates/router/Cargo.toml`][router-manifest] file.
+
+Of these features, the noteworthy ones are the `default` and `release` features:
+
+- `default`: This feature set enables a basic set of necessary features for both
+  development and production environments, such as the transactional APIs,
+  APIs used by the control center, Stripe compatibility, automatic payment
+  retries, in-memory caching, etc.
+
+- `release`: This feature set enables some additional features that are suitable
+  for production environments, such as AWS KMS integration, AWS S3 integration,
+  AWS SES integration, etc.
+
+Refer to the [documentation on cargo features][cargo-features] to understand how
+to select features with Cargo, when building the application server.
+
+## Building with the Dockerfile
+
+The Docker images for the application server and other components can be built
+using the [`Dockerfile`][dockerfile] using commands like so, substituting the
+Docker image tags with suitable values:
+
+- router:
+
+  ```shell
+  docker build \
+    --load \
+    --file Dockerfile \
+    --build-arg "BINARY=router" \
+    --tag hyperswitch-router \
+    .
+  ```
+
+- consumer:
+
+  ```shell
+  docker build \
+    --load \
+    --file Dockerfile \
+    --build-arg "BINARY=scheduler" \
+    --build-arg "SCHEDULER_FLOW=consumer" \
+    --tag hyperswitch-consumer \
+    .
+  ```
+
+- producer:
+
+  ```shell
+  docker build \
+    --load \
+    --file Dockerfile \
+    --build-arg "BINARY=scheduler" \
+    --build-arg "SCHEDULER_FLOW=producer" \
+    --tag hyperswitch-producer \
+    .
+  ```
+
+- drainer:
+
+  ```shell
+  docker build \
+    --load \
+    --file Dockerfile \
+    --build-arg "BINARY=drainer" \
+    --tag hyperswitch-drainer \
+    .
+  ```
+
+When our Docker images are built using the [`Dockerfile`][dockerfile], the
+`cargo` command being run is:
+
+```shell
+cargo build --release --features release ${EXTRA_FEATURES}
+```
+
+- The `--release` flag specifies that optimized release binaries must be built.
+  Refer to the [`cargo build` manual page][cargo-build-manual-page] for more
+  information.
+
+- The `--features release` flag specifies that the `release` feature set must be
+  enabled for the build.
+  Since we do not specify the `--no-default-features` flag to the `cargo build`
+  command, the build would have the `default` and `release` features enabled.
+
+- The `${EXTRA_FEATURES}` build argument can specify any additional features
+  that would have to be passed to the `cargo build` command.
+  The build argument could look like so:
+  `EXTRA_FEATURES="--features feature1,feature2,feature3"`, with actual feature
```

Comment:
```suggestion
  `EXTRA_FEATURES=",feature1,feature2,feature3"`, with actual feature
```
the dockerfile uses this env as follows 
`RUN cargo build --release --features release ${EXTRA_FEATURES}`

Reply:
The `--features` flag can be specified multiple times, not a problem. And the feature names cannot be space separated unless we use quotes.